modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala

```diff
@@ -189,6 +189,139 @@ private[elasticsearch] final case class Exists[S](field: String) extends ExistsQ
+   * specifies how the computed scores are combined
+   *
+   * @param value
+   *   a value that we want to set `scoreMode` to, possible values are:
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.Avg]]
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.First]]
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.Max]]
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.Min]]
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.Multiply]]
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.None]]
+   *   - [[zio.elasticsearch.query.FunctionScoreScoreMode.Sum]]
+   * @return
+   *   an instance of [[zio.elasticsearch.query.FunctionScoreQuery]] enriched with the `scoreMode` parameter.
+   */
+  def scoreMode(value: FunctionScoreScoreMode): FunctionScoreQuery[S]
+
+  /**
+   * Adds one or multiple [[zio.elasticsearch.query.FunctionScoreFunction]] to existing
+   * [[zio.elasticsearch.query.FunctionScore]] query.
+   *
+   * @param functions
+   *   multiple [[zio.elasticsearch.query.FunctionScoreFunction]] to be added to query
+   * @tparam S1
+   *   the type of the sub-queries, for which an implicit [[zio.schema.Schema]] is required
+   * @return
+   *   an instance of [[zio.elasticsearch.query.FunctionScoreQuery]] enriched with the `functionScoreFunctions`
+   *   parameter.
+   */
+  def withFunctions[S1 <: S: Schema](functions: FunctionScoreFunction[S1]*): FunctionScoreQuery[S1]
+}
+
+private[elasticsearch] final case class FunctionScore[S](
+  functionScoreFunctions: Chunk[FunctionScoreFunction[S]],
+  boost: Option[Double],
+  boostMode: Option[FunctionScoreBoostMode],
+  maxBoost: Option[Double],
+  minScore: Option[Double],
+  query: Option[ElasticQuery[S]],
+  scoreMode: Option[FunctionScoreScoreMode]
+) extends FunctionScoreQuery[S] { self =>
+
+  def boost(value: Double): FunctionScoreQuery[S] =
+    self.copy(boost = Some(value))
+
+  def boostMode(value: FunctionScoreBoostMode): FunctionScoreQuery[S] =
+    self.copy(boostMode = Some(value))
+
+  def maxBoost(value: Double): FunctionScoreQuery[S] =
+    self.copy(maxBoost = Some(value))
+
+  def minScore(value: Double): FunctionScoreQuery[S] =
+    self.copy(minScore = Some(value))
+
+  def query(value: ElasticQuery[S]): FunctionScoreQuery[S] =
+    self.copy(query = Some(value))
+
+  def scoreMode(value: FunctionScoreScoreMode): FunctionScoreQuery[S] =
+    self.copy(scoreMode = Some(value))
+
+  def withFunctions[S1 <: S: Schema](functions: FunctionScoreFunction[S1]*): FunctionScoreQuery[S1] =
+    self.copy(functionScoreFunctions = functionScoreFunctions ++ functions)
+
+  private[elasticsearch] def toJson(fieldPath: Option[String]): Json =
+    Obj(
+      "function_score" -> Obj(
+        Chunk(
+          Some("functions" -> Arr(functionScoreFunctions.map(_.toJson))),
+          boost.map("boost" -> Num(_)),
+          boostMode.map(bm => "boost_mode" -> Str(s"${bm.toString.toLowerCase}")),
```

Comment:
```suggestion
          boostMode.map(bm => "boost_mode" -> Str(bm.toString.toLowerCase)),
```

Reply:
Can you also change `Str(bm.toString.toLowerCase))` to bm.toString.toLowerCase.toJson?
Also change this in other places, if any.